concepts/lists/introduction.md

```diff
@@ -0,0 +1,182 @@
+# Introduction
```

Comment:
This introduction document is quite large. As per [the docs](https://exercism.org/docs/building/tracks/concepts), we want it to be as minimal as possible.
That said, there is no exercise yet that teaches this concept, so it is not yet possible to determine the right level of detail.